throw.go

```diff
@@ -0,0 +1,45 @@
+package errors
+
+import "strings"
+
+// ApplicationError an appliction error with predifined error variable and detail message
+type ApplicationError struct {
+	// Inner inner error
+	Inner error
+	// MsgList detail message
+	MsgList []string
+}
+
+// Error implement error.Error
+func (e *ApplicationError) Error() string {
+	if e == nil {
+		return ""
+	}
+
+	if e.Inner != nil && e.MsgList != nil {
+		return e.Inner.Error() + ": " + strings.Join(e.MsgList, ", ")
+	} else if e.Inner != nil {
+		return e.Inner.Error()
+	} else {
+		return strings.Join(e.MsgList, ", ")
+	}
+
+}
+
+// Unwrap implement error.Unwrap
+func (e *ApplicationError) Unwrap() error {
+	if e == nil {
+		return nil
+	}
+	return e.Inner
+}
+
+// Throw create an application error with prefinded error variable and message
+// example
+//    errors.Throw(ErrInvalidParameter, "bloober_id is missing")
+func Throw(inner error, msgList ...string) error {
```

Comment:
Here we are actually wrapping an error with msgs. should it not be `Wrapf` 🤔 

with `Wrap` also this can be achieved this way.
```
errors.Wrap(err, fmt.Sprintf("%s,%s,%s, msg1, msg2, msg3))
or 
errors.Wrap(err, strings.Join([]string{msg1, msg2, msg3}, ",")) 
```

If we want to accept multiple params, that be done with the `Wrapf` function right.

Reply:
Throw is implemented based on standard go error package without any own protocol/interface.

Reply:
so it can work with any package without any changes.

Reply:
for msgList, it is not for format, it is working with many errors. for example, there are many failed request from sharder api.